### PR TITLE
`General`: Improve CSV header parsing when importing users into course groups

### LINKED ISSUE #4939: Importing students into a course is failing
### Describe the bug

Importing students from the Tum Online CSV file is not possible. 

### To Reproduce

1. Go to any course
2. Click on import students 
3. Select the CSV file 
4. Check that you have a file with the total number of students
5. Import it and check if you get a message at the end of the dialogue that says Not found: # students 

### Expected behavior

Should be import it the list of students

### Screenshots

image

### What browsers are you seeing the problem on?

Chrome

### Additional context

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by maximiliansoelch**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

This is a follow-up PR to the fix [b054013](https://github.com/ls1intum/Artemis/commit/b054013ff172ee355d41e1fe6c11e474cffaa62e), which closed #4939.
The changes introduced in this PR simplify the code and also make it clearer where new header fields need to be added in the future.
In addition, this PR addresses an additional improvement mentioned in a comment in [#4939](https://github.com/ls1intum/Artemis/issues/4939#issuecomment-1106570075): 
> All "not found" lines should be red, not only every second one
This refers to the text color.

### Description

This PR transitions from a single object, which previously stored the possible csv header fields in the object's parameters, to several arrays.

### Steps for Testing

Prerequisites:
- 1 Course

1. Log in to Artemis as Admin
2. Navigate to Course Administration
3. Select a course and go to the student overview
4. Click Import Users
5. Select the CSV file

- [invalid_rows_2_and_6.csv](https://github.com/ls1intum/Artemis/files/8543832/invalid_rows_2_and_6.csv)
- [valid_test_3_of_5_users_imported.csv](https://github.com/ls1intum/Artemis/files/8543825/valid_test_3_of_5_users_imported.csv)

7. Click on Import
- for the invalid file an error message should be shown, indicating errors in row 2 and 6:
Bildschirmfoto 2022-04-22 um 22 26 55

- for the valid file 3 out of 5 users should get imported

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Branch | Line |
|------------|-------:|-----:|
| users-import-dialog.component.ts | 78.3% | 89.52% |

### Screenshots

Before the changes, only every second line the import status was applied:
Bildschirmfoto 2022-04-22 um 22 15 08

After the changes the text color corresponding to the import status is applied correctly regardless of the row position:
Bildschirmfoto 2022-04-22 um 22 18 43

### CODE REVIEW

**Review by ge65cer — APPROVED** [review]
Tested with various valid/invalid CSV files and worked correctly. 👍

**Review by Santia-go — DISMISSED** [review]
Tested in TS1. 
Working as expected. It is possible to import the groups from the CSV.

File: src/main/webapp/app/shared/import/users-import-dialog.component.ts [hunk 1]
```diff
@@ -172,24 +134,30 @@ export class UsersImportDialogComponent implements OnDestroy {
...
+     * @param entry which should be checked if it contains one of the keys.
+     * @param keys that should be checked for in the entry.
+     */
+    checkIfEntryContainsKey(entry: CsvUser, keys: string[]): boolean {
+        for (const key of keys) {
+            if (entry[key] !== undefined && entry[key] !== '') {
+                return true;
+            }
+        }
+        return false;
```
Comment (bassner) [inline src/main/webapp/app/shared/import/users-import-dialog.component.ts:148]:
```suggestion
        return keys.some((key) => entry[key] && entry[key] !== '');
```

**Review by bassner — CHANGES_REQUESTED** [review]
Tested this on TS1 and it works as described 👍 

Just one remark :)

**Review by bassner — APPROVED** [review]
Thanks for applying the feedback. LGTM! 👍

**Review by ge65cer — APPROVED** [review]

**Review by max-bergmann — APPROVED** [review]
Tested on TS3. Works as expected.

**Review by Santia-go — APPROVED** [review]
Tested on TS1
Re-approved!

**Pull request merged**